plugins/modules/monitoring/spectrum_model_attrs.py

```diff
@@ -0,0 +1,526 @@
+                </equals>
+            </and>
+        </filtered-models>
+        """
+
+        # Parent filter tag
+        filtered_models = ET.Element('filtered-models')
+        # Logically and
+        _and = ET.SubElement(filtered_models, 'and')
+
+        # Model Name
+        MN_equals = ET.SubElement(_and, 'equals')
+        Model_Name = ET.SubElement(MN_equals, 'attribute',
+                                   {'id': self.attr_map['Model_Name']})
+        MN_value = ET.SubElement(Model_Name, 'value')
+        MN_value.text = mname
+
+        # Model Type Name
+        MTN_equals = ET.SubElement(_and, 'equals')
+        Modeltype_Name = ET.SubElement(MTN_equals, 'attribute',
+                                       {'id': self.attr_map['Modeltype_Name']})
+        MTN_value = ET.SubElement(Modeltype_Name, 'value')
+        MTN_value.text = mtype
+
+        return self.find_model(ET.tostring(filtered_models,
+                                           encoding='unicode'),
+                               ret_attrs)
+
+    def ensure_model_attrs(self):
+
+        # Get a list of all requested attribute names/IDs plus Model_Handle and
+        # use them to query the values currently set. Store finding in a
+        # dictionary.
+        req_attrs = []
+        for attr in self.module.params['attributes']:
+            req_attrs.append(attr['name'])
+        if 'Model_Handle' not in req_attrs:
+            req_attrs.append('Model_Handle')
+
+        # Survey attributes currently set and store in a dict.
+        cur_attrs = self.find_model_by_name_type(self.module.params['name'],
+                                                 self.module.params['type'],
+                                                 req_attrs)
+
+        # Iterate through the requested attributes names/IDs values pair and
+        # compare with those currently set. If different, attempt to change.
+        Model_Handle = cur_attrs.pop("Model_Handle")
+        for attr in self.module.params['attributes']:
+            req_name = attr['name']
+            req_val = attr['value']
+            if req_val == "":
+                # The API will return None on empty string
+                req_val = None
+            if cur_attrs[req_name] != req_val:
+                if self.module.check_mode:
+                    self.result['changed_attrs'][req_name] = req_val
+                    self.result['msg'] = self.success_msg
+                    self.result['changed'] = True
+                    continue
+                resp = self.update_model(Model_Handle, {req_name: req_val})
+
+        self.module.exit_json(**self.result)
+
+
+def run_module():
+    argument_spec = dict(
+        url=dict(type='str', required=True),
+        url_username=dict(type='str', required=True, aliases=['username']),
+        url_password=dict(type='str', required=True, aliases=['password'],
+                          no_log=True),
+        validate_certs=dict(type='bool', default='yes'),
```

Comment:
```suggestion
        validate_certs=dict(type='bool', default=True),
```
`'yes'` is a string, not a boolean. Same below.